### ISSUE TITLE
Issues with Next.js

### ISSUE THREAD

**Issue by wesbos** [kybarg/react-qr-scanner#38]
Hello - I think this fork does SSR, right? 

When I import it like so:

```js
import QrReader from 'react-qr-scanner';
```

I get this issue: `SyntaxError: Cannot use import statement outside a module`  pointing to `> 1 | module.exports = require("react-qr-scanner");` 

I looked through the source and can't seem to find why it would think it's cjs. Any ideas? 

It does work if I use next.js dynamic import, but then that that annoy as it re-imports it when it re-renders. 

```
const QrReader = dynamic(() => import('react-qr-scanner'));
```

**Comment by wesbos** [comment]
I think this has to do with the Module not being shipped as CJS, or with a .MJS extension?

**Comment by AmYre** [comment]
Hi there,

Got the same issue here, with a slight difference. It is pointing to `\node_modules\react-qr-scanner\lib\index. `.  Dynamic import in that file, for the five imports did the trick for now. Thank you Wes "Barracuda" Bos ;)

**Comment by csellis** [comment]
Did you make any progress @wesbos ? I'm looking at this same use case

**Comment by wesbos** [comment]
Yes, I used an alternative lazy import package for React.I forget the name, but it was the big one...

**Comment by theabdulmateen** [comment]
> Yes, I used an alternative lazy import package for React.I forget the name, but it was the big one...

By this do you mean wait till the page is delivered from nextjs and then load the `QrReader`? In that case you would only need to something of this sort right?

```javascript
useEffect(() => {
  setHasRendered(true)
}, [])
...
...

<div>
{hasRendered && (
    <QrReader
      ...
    />
  )}
</div>
```

**Comment by ninest** [comment]
Apart from the above solution, this works for me

```tsx
<div>
  {typeof window !== "undefined" && (
    <>
      <QRReader
        onScan={onScan}
        onError={onError}
      ></QRReader>
    </>
  )}
</div>
```
